### ISSUE TITLE
jsx-indent-props causes wacky indentation when prop is a multiline object literal

### ISSUE THREAD

**Issue by bhallstein** [yannickcr/eslint-plugin-react#3004]
Having the [jsx-indent-props rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-indent-props.md) enabled and set to "first" will cause this:

```jsx
<CSelectFontFamily style={{
                     maxWidth: "6rem",
                     paddingTop: "0.15em",
                     paddingBottom: "0.15em",
                   }}
                   cssPropName={`--${mode}FontFamily-` + typographyNameLower}
                   name={typographyName + "Font"}
                   label="Font Family" />
```

To be reformatted into this:

```jsx
<CSelectFontFamily style={{
    maxWidth: "6rem",
    paddingTop: "0.15em",
    paddingBottom: "0.15em",
  }}
                   cssPropName={`--${mode}FontFamily-` + typographyNameLower}
                   name={typographyName + "Font"}
                   label="Font Family" />
```

Should this rule support smarter indent of object literals in props?

**Comment by ljharb** [comment]
If you have the `jsx-indent` and `indent` rules enabled, is that handled?

Although I'd never use that style (nor would any popular styleguide), I'd definitely expect the first example to be correct.

**Comment by bhallstein** [comment]
Thanks for the reply @ljharb. Yeah I agree it seems definitely incorrect that the second example should result from the `first` rule. I put it forward as a bug as on balance I felt it was unintended behaviour.

I can't speak for all styleguides but the style in the first code snippet is directly supported in this plugin via `jsx-indent-props`'s `first` setting. I'm sure everyone would agree that if the setting exists it should work well.

When I have a sec I'll do some more digging and add a reproducible test case or two, sorry that was missing from the original submission.

**Comment by ljharb** [comment]
@bhallstein any update?
